README.md
<a href="https://www.buymeacoffee.com/rodrigobastosv" target="_blank"><img src="https://cdn.buymeacoffee.com/buttons/v2/default-yellow.png" alt="Buy Me A Coffee" style="height: 60px !important;width: 217px !important;" ></a>

#  Introduction

Tag is a widget that simplify the process of creating a tag for any given widget. Just pass the string of the tag and Tag widget will position the tag above the desired widget.

# Basic Usage

The most simple usage is just wrap the widget that you want to be tagged in a Tag widget.

```dart
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        body: Center(
          child: Tag(
            child: Container(
              color: Colors.red,
              height: 200,
              width: 200,
            ),
            text: 'A red container',
            textStyle: TextStyle(
              color: Colors.white,
            ),
            color: Color(0xFFFF2D92),
            labelDecoration: BoxDecoration(
              borderRadius: BorderRadius.only(
                bottomLeft: Radius.circular(4),
                bottomRight: Radius.circular(4),
                topLeft: Radius.circular(4),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

This simple step will already configure the Tag overlay for use.

The example will produce something like this:

![enter image description here](https://i.ibb.co/NVnGShm/1.png)

You can use Tag widget with any widget of your choice. Here we have another example using a CircleAvatar.

```
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        body: Center(
          child: Tag(
            rightPadding: 12,
            child: CircleAvatar(
              backgroundImage: NetworkImage(
                  'https://t1.gstatic.com/images?q=tbn:ANd9GcSZqF3OGhXYqL8DSbLynQ-qA2EIniPAljZXhUTZ8Ay6f6g-um21wsmJsBkeLkjW'),
            ),
            text: 'Lewis Hamilton',
            textStyle: TextStyle(
              color: Colors.white,
            ),
            color: Colors.grey,
            labelDecoration: BoxDecoration(
              borderRadius: BorderRadius.only(
                bottomLeft: Radius.circular(4),
                bottomRight: Radius.circular(4),
                topLeft: Radius.circular(4),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

And produces this:

![enter image description here](https://i.ibb.co/yXtrk17/2.png)

# Customisation
There are many types of configurations that can be done to the Tag widget. You can configure spacing, tag decoration and much more. Just check the Tag constructor to see all possibilities. Missing any customisation? Please file a bug and i will be more than happy to work on it.

## Todo

- [ ] Tests
- [ ] Add more options to the tip of the Tag

## Suggestions & Bugs

For any suggestions or bug report please head to [issue tracker][tracker].

[tracker]: https://github.com/rodrigobastosv/tag/issues


CHANGELOG.md
# 2.0.0+2
- Adjusting README

# 2.0.0+1
- Small adjusts
- 
# 2.0.0
- Migrating to null safe

# 1.2.0

- Making API simpler and more reliable

# 1.0.1+1

- Fixing more bugs

# 1.0.1

- Fixing bugs

# 1.0.0

- Initial release fully functional

lib/src/tag.dart
import 'package:flutter/material.dart';

import 'clippers/triangle_clipper.dart';

class Tag extends StatelessWidget {
  Tag({
    Key? key,
    required this.child,
    required this.text,
    this.textStyle,
    this.labelDecoration,
    this.color,
    this.tagHeight = 40,
    this.tagWidth = 120,
    this.cornerHeight = 10,
    this.cornerWidth = 10,
    this.spacing = 2,
    this.rightPadding = 0,
    this.isVisible = true,
  }) : super(key: key);

  /// Widget that will receive the tag
  final Widget child;

  /// Text to go on the tag if any
  final String text;

  /// Style of the text
  final TextStyle? textStyle;

  /// Decoration of the text label
  final BoxDecoration? labelDecoration;

  /// Color of the tag
  final Color? color;

  /// Height of the tag
  final double tagHeight;

  /// Width of the tag
  final double tagWidth;

  /// Height of the corner
  final double cornerHeight;

  /// Width of the corner
  final double cornerWidth;

  /// Spacing of the tag text
  final double spacing;

  /// Padding
  final double rightPadding;

  /// Wheter the tag is visible or not
  final bool isVisible;

  @override
  Widget build(BuildContext context) {
    return Stack(
      clipBehavior: Clip.none,
      children: [
        child,
        if (isVisible)
          Positioned(
            right: rightPadding,
            bottom: -spacing - cornerHeight - tagHeight,
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.end,
              children: [
                ClipPath(
                  clipper: TriangleClipper(),
                  child: Container(
                    color: color,
                    height: cornerHeight,
                    width: cornerWidth,
                  ),
                ),
                Container(
                  decoration: labelDecoration?.copyWith(
                    color: color,
                  ),
                  height: tagHeight,
                  width: tagWidth,
                  child: Center(
                    child: Text(
                      text,
                      style: textStyle,
                    ),
                  ),
                ),
              ],
            ),
          ),
      ],
    );
  }
}


example/lib/a.dart
import 'package:flutter/material.dart';
import 'package:tag/tag.dart';

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        body: Center(
          child: Tag(
            rightPadding: 12,
            child: CircleAvatar(
              backgroundImage: NetworkImage(
                  'https://t1.gstatic.com/images?q=tbn:ANd9GcSZqF3OGhXYqL8DSbLynQ-qA2EIniPAljZXhUTZ8Ay6f6g-um21wsmJsBkeLkjW'),
            ),
            text: 'Lewis Hamilton',
            textStyle: TextStyle(
              color: Colors.white,
            ),
            color: Colors.grey,
            labelDecoration: BoxDecoration(
              borderRadius: BorderRadius.only(
                bottomLeft: Radius.circular(4),
                bottomRight: Radius.circular(4),
                topLeft: Radius.circular(4),
              ),
            ),
          ),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:tag/tag.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: Scaffold(
        body: Center(
          child: Tag(
            child: Container(
              color: Colors.red,
              height: 200,
              width: 200,
            ),
            text: 'A red container',
            textStyle: TextStyle(
              color: Colors.white,
            ),
            color: Color(0xFFFF2D92),
            labelDecoration: BoxDecoration(
              borderRadius: BorderRadius.only(
                bottomLeft: Radius.circular(4),
                bottomRight: Radius.circular(4),
                topLeft: Radius.circular(4),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
